Better handling of Dynamic modules and (re)loading

JorisGoosen opened the pull request:
- Removed some stuff from Analysis/Analyses for oldstyle modules, as we don't have them anymore
- Made sure reloading Development Module does not crash anymore
-- fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/1175
-- This was solved by remembering the _slaveFinishedConnection = connect(_slaveProcess, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),	this, &EngineRepresentation::processFinished);
-- And removing it when an engine stopped, a race condition occured whereby the new engine started so fast it got the "closed" signal from the previous one and interpreted it as a crash. While in fact it worked fine.
- Added a loading indicator in the ribbonbar for modules while they are loading
- Added an error indicator, but Im not sure in which situation a module would have an error *and remain installed/loaded*. So not actually sure if it makes sense or not. I guess it would be useful in case an icon is missing. But a then a question mark would make more sense.
- Added a few more logging statements to engine startup
- Removed the "not running analyses on engine 0" define in the constructor of EngineRepresentation...

boutinb commented on Desktop/engine/enginerepresentation.cpp:
```diff
@@ -23,7 +18,7 @@ void EngineRepresentation::setSlaveProcess(QProcess * slaveProcess)
 	_slaveProcess->setParent(this);
 
-	connect(_slaveProcess, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),	this, &EngineRepresentation::processFinished);
+	_slaveFinishedConnection = connect(_slaveProcess, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),	this, &EngineRepresentation::processFinished);
```
Here you make a copy of the connection: https://stackoverflow.com/questions/14828678/disconnecting-lambda-functions-in-qt5
This won't work.

JorisGoosen replied:
So, why does it solve the problem?

Also see: https://doc.qt.io/qt-5/qmetaobject-connection.html
It supports both move and copy constructors.

Also it is specifically the object that gets returned by `connect`: https://doc.qt.io/qt-5/qobject.html#connect
And received by `disconnect`: https://doc.qt.io/qt-5/qobject.html#disconnect-4 

Furthermore, in the linked documentation the following is stated literally:
> The function returns a QMetaObject::Connection that represents a handle to a connection if it successfully connects the signal to the slot. The connection handle will be invalid if it cannot create the connection, for example, if QObject is unable to verify the existence of either signal or method, or if their signatures aren't compatible. You can check if the handle is valid by casting it to a bool.

Unless you can actually show me that it doesn't work beyond some vague SO post from a long time ago I won't make any change. 

<img width="468" alt="afbeelding" src="https://user-images.githubusercontent.com/29062713/100632005-9e290c80-332c-11eb-885b-0fbe09c47fe9.png">
Not to mention the fact that it is about `lambda connects` and not about a normal QObject to QObject connect.

boutinb replied:
OK. False alarm apparently.

boutinb commented:
@JorisGoosen There are now conflicts with submodules...

JorisGoosen commented:
I'll rebase it!

JorisGoosen committed:
Better handling of Dynamic modules and (re)loading
- Removed some stuff from Analysis/Analyses for oldstyle modules, as we don't have them anymore
- Made sure reloading Development Module does not crash anymore
-- fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/1175
-- This was solved by remembering the _slaveFinishedConnection = connect(_slaveProcess, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),	this, &EngineRepresentation::processFinished);
-- And removing it when an engine stopped, a race condition occured whereby the new engine started so fast it got the "closed" signal from the previous one and interpreted it as a crash. While in fact it worked fine.
- Added a loading indicator in the ribbonbar for modules while they are loading
- Added an error indicator, but Im not sure in which situation a module would have an error *and remain installed/loaded*. So not actually sure if it makes sense or not. I guess it would be useful in case an icon is missing. But a then a question mark would make more sense.
- Added a few more logging statements to engine startup
- Removed the "not running analyses on engine 0" define in the constructor of EngineRepresentation...

JorisGoosen commented:
This should be better.

JorisGoosen commented:
Can you make sure to "Rebase and merge" instead of the other options, that way our commithistory looks nicer.
